pkg/testing/builder/pod.go

```diff
@@ -105,9 +105,30 @@ func (p *PodBuilder) WithRequiredDuringSchedulingIgnoredDuringExecution() *PodBu
 	return p
 }
 
+func (p *PodBuilder) WithPreferredDuringSchedulingIgnoredDuringExecution(opts ...func(*v1.PreferredSchedulingTerm)) *PodBuilder {
+	p.WithNodeAffinity()
+
+	if p.pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution == nil {
+		p.pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution = []v1.PreferredSchedulingTerm{}
+	}
+
+	// Apply each optional function
+	for _, opt := range opts {
+		term := v1.PreferredSchedulingTerm{}
+		opt(&term)
+		p.pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution = append(
+			p.pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution,
+			term,
+		)
+	}
+
+	return p
+}
```

Comment:
```suggestion
func (p *PodBuilder) WithPreferredDuringSchedulingIgnoredDuringExecution(values ...*v1.PreferredSchedulingTerm) *PodBuilder {
	p.WithNodeAffinity()

	if p.pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution == nil {
		p.pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution = []v1.PreferredSchedulingTerm{}
	}

	for i := range values {
		if values[i] == nil {
			panic("nil value passed to WithPreferredDuringSchedulingIgnoredDuringExecution")
		}
		p.pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution = append(
			p.pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution, *values[i])
	}
	return p
}
```
can this func just pass `*v1.PreferredSchedulingTerm` directly instead of `func(*v1.PreferredSchedulingTerm)`?